### DISCUSSION TITLE
Is there a way to have more than one instance of the upload widget?

### DISCUSSION THREAD

**Discussion by krnrnee** [cloudinary-community/next-cloudinary#152] [Q&A] [answered]
I need two distinct sets of behavior after upload, but since the widget is stored in memory the onUpload function is set the first time I call the widget.

**Comment by colbyfayock** [comment, marked as answer]
hey @krnrnee i _think_ it should work as is? are you using the latest version? its possible this was fixed while I was addressing another bug, but i just tried in the docs examples adding a second unsigned widget with 2 different console logs and it seems to handle both okay

<img width="954" alt="image" src="https://user-images.githubusercontent.com/1045274/222468788-0ed88508-b93f-40df-ab65-bb709a08bdcb.png">

let me know if thats not what you mean or if you're seeing a different issue

**Reply by krnrnee** [reply]
Hi @colbyfayock, thanks for the quick response.  I will make sure I am using the latest version and let you know if that works.  One thing however, I am using a signed upload rather than an unsigned upload.  Should that also work in the latest version?

**Reply by krnrnee** [reply]
@colbyfayock - I updated to the latest version and that seems to have fixed it!  Thanks!

**Reply by colbyfayock** [reply]
yay awesome! glad to hear that
